### PR TITLE
Added csrf protection support

### PR DESCRIPTION
**Pull request by dschwar**
Added config option to enable CSRF headers and to send "csrf-token" in the “X-CSRF-Token” HTTP header when making requests.

### CODE REVIEW

File: app/views/graphiql/rails/editors/show.html.erb [hunk 1]
```diff
@@ -4,6 +4,9 @@
     <title>GraphiQL</title>
     <%= stylesheet_link_tag("graphiql/rails/application") %>
     <%= javascript_include_tag("graphiql/rails/application") %>
+    <% if GraphiQL::Rails.config.csrf %>
+    <%= csrf_meta_tags %>
```
Comment (rmosolgo) [inline app/views/graphiql/rails/editors/show.html.erb]: Do we still need the meta tag if we're sending the token by request header? (I _thought_ the headers were just so JS could add it to request headers)

**Comment by rmosolgo** [comment]
Nice! could you add a line to the README, too? You can put it under the other config notes

File: app/views/graphiql/rails/editors/show.html.erb (hunk 1 above)
Comment (dschwar) [inline app/views/graphiql/rails/editors/show.html.erb]: Didn't realize at first that `form_authenticity_token` will use it's own token. Changed to use javascript to get the meta tag with the token.

**Comment by rmosolgo** [comment]
Thanks, that's awesome!

**Pull request merged**